### DISCUSSION TITLE
Stuck after  searx      | [uwsgi-static] added mapping for /static => /usr/local/searx/searx/static

### DISCUSSION THREAD

**Discussion by ghost** [searx/searx#2750] [Installation issue: Docker]
Hello,

Really lost and would appreciate any help.

I'm using a fresh Debian VM with 1 core and 2gb of  memory under Proxmox. I followed the instruction completely, however, I'm stuck after sudo ./start.sh. It just hangs there and I have no idea how to troubleshoot this... 

Stuck after` searx | [uwsgi-static] added mapping for /static => /usr/local/searx/searx/static`.

The CPU at this time is at ~3% and memory is at ~90%. Doesn't look like a resource problem (?)

Would anyone knows what's going on?

```
use docker-compose.yaml
Removing morty   ... done
Removing caddy   ... done
Removing searx   ... done
Removing filtron ... done
Removing network searx-docker_searx
No stopped containers
Creating network "searx-docker_searx" with the default driver
Creating caddy   ... done
Creating morty   ... done
Creating filtron ... done
Creating searx   ... done
Attaching to caddy, searx, filtron
caddy      | {"level":"info","ts":1618137300.5127275,"msg":"using provided configuration","config_file":"/etc/caddy/Caddyfile","config_adapter":"caddyfile"}
caddy      | {"level":"warn","ts":1618137300.5501466,"logger":"admin","msg":"admin endpoint disabled"}
caddy      | {"level":"info","ts":1618137300.5676029,"logger":"tls.cache.maintenance","msg":"started background certificate maintenance","cache":"0xc000216cb0"}
caddy      | {"level":"info","ts":1618137300.5875523,"logger":"http","msg":"server is listening only on the HTTPS port but has no TLS connection policies; adding one to enable TLS","server_name":"srv0","https_port":443}
caddy      | {"level":"info","ts":1618137300.5875962,"logger":"http","msg":"enabling automatic HTTP->HTTPS redirects","server_name":"srv0"}
caddy      | {"level":"warn","ts":1618137300.801205,"logger":"pki.ca.local","msg":"installing root certificate (you might be prompted for password)","path":"storage:pki/authorities/local/root.crt"}
caddy      | 2021/04/11 10:35:00 Warning: "certutil" is not available, install "certutil" with "apt install libnss3-tools" or "yum install nss-tools" and try again
caddy      | 2021/04/11 10:35:00 define JAVA_HOME environment variable to use the Java trust
caddy      | 2021/04/11 10:35:00 certificate installed properly in linux trusts
caddy      | {"level":"info","ts":1618137300.9953613,"logger":"http","msg":"enabling automatic TLS certificate management","domains":["localhost"]}
caddy      | {"level":"warn","ts":1618137300.9962409,"logger":"tls","msg":"stapling OCSP","error":"no OCSP stapling for [localhost]: no OCSP server specified in certificate"}
caddy      | {"level":"info","ts":1618137300.996499,"msg":"autosaved config","file":"/config/caddy/autosave.json"}
caddy      | {"level":"info","ts":1618137300.9965127,"msg":"serving initial configuration"}
caddy      | {"level":"info","ts":1618137301.003832,"logger":"tls","msg":"cleaned up storage units"}
filtron    | 2021/04/11 10:35:02 15 rules loaded from /etc/filtron/rules.json
filtron    | 2021/04/11 10:35:02 API listens on 0.0.0.0:4041
filtron    | 2021/04/11 10:35:02 Proxy listens on 0.0.0.0:4040
searx      | searx version 1.0.0-48-5d5ecdb7
searx      |
searx      | Create /etc/searx/uwsgi.ini
searx      | Create /etc/searx/settings.yml
searx      | Listen on 0.0.0.0:8080
searx      | [uWSGI] getting INI configuration from /etc/searx/uwsgi.ini
searx      | [uwsgi-static] added mapping for /static => /usr/local/searx/searx/static
```

**Comment by dalf** [comment]
`./start.sh` doesn't start the process in background.

See: https://github.com/searx/searx-docker/blob/4ad61d64991d76a7d300b7b867771ffed9f224ca/start.sh#L11

If you have changed the values of `SEARX_HOSTNAME` in `.env`, go to `https://${SEARX_HOSTNAME}`.

Use `docker-compose up -d` to start everything in background.

**Comment by ghost** [comment]
I’m sorry I don’t think I understand. So ./start.sh is just for checking only right? I cannot pass that check so I’m assuming something is wrong.

just tried docker-compose up -d and cannot access the search page via local host.

docker-compose up shows the same error, stuck at searx      | [uwsgi-static] added mapping for /static => /usr/local/searx/searx/static
It should normally move on to the next step right?

The only thing I’ve changed in .env is the morty key.

(Perhaps the problem is Debian-specific? I also tried bare metal install but didn’t work also.)

Anyway thanks for your reply

**Reply by ghost** [reply]
I notice that when I go to http://localhost it redirects to https://localhost. So I think the caddy server is working somehow. But it’s a blank page

**Reply by mlysle** [reply]
I experienced this issue.
If you expose the ports, you should be able to connect. I added:
>   searx:
> ...
>     ports:
>       - "127.0.0.1:8080:8080"
> 
and now it's working. :)
I also had to tweak firefox's security settings to connect to a web server on a non-standard port. I recommend testing with curl.

And no, it's not a distro specific issue since I use a completely unrelated distribution.
